src/utils.c

```diff
@@ -123,6 +123,21 @@ char *get_album_cover_file(const char *location, const char *format) {
 	regex_t regex;
 	char *tmp;
 
+	/* chek cue file path play  */
+	const char patern[7] = "cue://";
+	char *location_t;
+	location_t = strdup(location);
```

Comment:
What if the `location` is NULL? Also, the `location_t` is not freed.

You should place this code after `location` NULL check at line 128. Simply check for "cue://" at the beginning of the `location` string and move the pointer 6 characters forward. It should be 2-3 added lines at most :)